src/sedcli_main.c

```diff
@@ -329,10 +329,32 @@ static int get_password(struct sed_key *pwd)
 	return 0;
 }
 
+static void sed_status_ret(int status, struct sed_device *dev)
+{
+	const char *sed_status = NULL;
+
+	sed_status = sed_error_text(status);
+
+	if (sed_status == NULL && status != 0) {
+		if (status == -EINVAL) {
+			sedcli_printf(LOG_ERR, "Invalid parameter\n");
+		} else if (status == -ENODEV) {
+			sedcli_printf(LOG_ERR, "Couldn't determine device state\n");
+		} else if (status == -ENOMEM) {
+			sedcli_printf(LOG_ERR, "No memory\n");
+		} else {
+			sedcli_printf(LOG_ERR, "Unknown error\n");
+		}
+	 } else {
+		 sedcli_printf(LOG_INFO, "%s\n", sed_status);
+	 }
+
+	sed_deinit(dev);
```

Comment:
sed_deinit doesn't logically fit in here, sed_status_ret function should be responsible for interpreting and printing an error and status, it should not be responsible for deinitialization of sed_device context.
Please remove sed_deinit and struct sed_device function argument because it doesn't belong here

Reply:
Or does it make sense to add (_and_deinit) to the function name.
Anyway we do deinit in every handle functions

Reply:
> Or does it make sense to add (_and_deinit) to the function name.
> Anyway we do deinit in every handle functions

I don't think it make sense. What if somebody decides to deallocate device context by himself and just print the message. Your function would require him to provide dev context...

Reply:
ok :)

Reply:
done